src/object/cli_obj.c

```diff
@@ -1277,147 +1257,129 @@ obj_iod_valid(unsigned int nr, daos_iod_t *iods, bool update)
 	return true;
 }
 
-int
-dc_obj_fetch(tse_task_t *task)
+/* check if the obj request is valid */
+static int
+obj_req_valid(void *args, int opc, daos_epoch_t *epoch)
 {
-	daos_obj_fetch_t	*args = dc_task_get_args(task);
-	struct obj_auxi_args	*obj_auxi;
-	struct dc_object	*obj;
-	struct dc_obj_shard	*obj_shard;
-	int			 shard;
-	unsigned int		 map_ver;
-	uint64_t		 dkey_hash;
-	daos_epoch_t		 epoch;
-	int			 rc;
-
-	if (args->dkey == NULL || args->dkey->iov_buf == NULL || args->nr == 0
-	    || !obj_iod_valid(args->nr, args->iods, false))
-		D_GOTO(out_task, rc = -DER_INVAL);
-
-	rc = dc_tx_check(args->th, false, &epoch);
-	if (rc) {
-		if (rc != -DER_INVAL)
-			goto out_task;
-		/* FIXME: until distributed transaction. */
-		epoch = DAOS_EPOCH_MAX; /* = daos_ts2epoch();*/
-		D_DEBUG(DB_IO, "set epoch "DF_U64"\n", epoch);
-	}
-	D_ASSERT(epoch);
+	int	rc = 0;
 
-	obj = obj_hdl2ptr(args->oh);
-	if (obj == NULL)
-		D_GOTO(out_task, rc = -DER_NO_HDL);
+	switch (opc) {
+		daos_obj_fetch_t	*f_args;
+		daos_obj_update_t	*u_args;
+		daos_obj_punch_t	*p_args;
 
-	obj_auxi = tse_task_stack_push(task, sizeof(*obj_auxi));
-	obj_auxi->opc = DAOS_OBJ_RPC_FETCH;
-	rc = tse_task_register_comp_cb(task, obj_comp_cb, &obj,
-				       sizeof(obj));
-	if (rc != 0) {
-		/* NB: process_rc_cb() will release refcount in other cases */
-		obj_decref(obj);
-		D_GOTO(out_task, rc);
-	}
+	case DAOS_OBJ_RPC_FETCH:
```

Comment:
(style)  Possible switch case/default not preceeded by break or fallthrough comment